tools/protovalidate-conformance/internal/cases/cases_message.go

```diff
@@ -155,5 +155,40 @@ func messageSuite() suites.Suite {
 				},
 			),
 		},
+		"oneof/single-field/valid": {
+			// BoolField is not part of the oneof, so it can be set on this message
+			Message:  &cases.MessageOneofSingleField{StrField: "test", BoolField: true},
+			Expected: results.Success(true),
+		},
+		"oneof/two-fields/valid": {
+			Message:  &cases.MessageOneofMultipleFields{StrField: "test"},
+			Expected: results.Success(true),
+		},
+		"oneof/two-fields/invalid": {
+			Message: &cases.MessageOneofMultipleFields{StrField: "test", BoolField: true},
+			Expected: results.Violations(
+				&validate.Violation{
+					RuleId:  proto.String("message.oneof"),
+					Message: proto.String("only one of str_field, bool_field can be set"),
+				},
+			),
+		},
+		"oneof/multiple-fields/required/valid": {
+			Message:  &cases.MessageOneofMultipleFieldsRequired{StrField: "test"},
+			Expected: results.Success(true),
+		},
+		"oneof/multiple-fields/required/invalid": {
+			Message: &cases.MessageOneofMultipleFieldsRequired{},
+			Expected: results.Violations(
+				&validate.Violation{
+					RuleId:  proto.String("message.oneof"),
+					Message: proto.String("one of str_field, bool_field must be set"),
+				},
+			),
+		},
+		"oneof/unknown-field/invalid": {
+			Message:  &cases.MessageOneofUnknownFieldName{},
+			Expected: results.CompilationError("field not found in message"),
```

Comment:
Nit: Why not include the full expected error message?

```suggestion
			Expected: results.CompilationError("field not found in message buf.validate.conformance.cases.MessageOneofUnknownFieldName"),
```

IIRC, the conformance tests don't actually check the message, but maybe we'll support that at some point. Either way, a failing test will print this expectation with the full error message.